internal/daemon/dpudaemon.go

```diff
@@ -125,96 +150,99 @@ func (d *DpuDaemon) Listen() (net.Listener, error) {
 		return d.cniCmdNfDelHandler(r)
 	}
 
-	d.cniserver = cniserver.NewCNIServer(add, del, cniserver.WithSocketPath(d.cniServerPath))
+	d.cniserver = cniserver.NewCNIServer(add, del, cniserver.WithSocketPath(d.pathManager.CNIServerPath()))
 
 	return lis, err
 }
 
 func (d *DpuDaemon) ListenAndServe() error {
-	var wg sync.WaitGroup
-	done := make(chan error, 1)
-
 	listener, err := d.Listen()
 
 	if err != nil {
 		d.log.Error(err, "Failed to listen")
 		return err
 	}
 
-	wg.Add(1)
+	return d.Serve(listener)
+}
+
+func (d *DpuDaemon) Serve(listener net.Listener) error {
+	d.wg.Add(3)
```

Comment:
It is better convention to Add(1) next to each go func for better traceability. I think

Reply:
done